sdmetrics/multi_table/statistical/cardinality_shape_similarity.py

```diff
@@ -0,0 +1,127 @@
+                table names and pandas.DataFrames.
+            synthetic_data (dict[str, pandas.DataFrame]):
+                The tables from the synthetic dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            metadata (dict):
+                Multi-table metadata dict. If not passed, it is build based on the
+                real_data fields and dtypes.
+
+        Returns:
+            dict:
+                A dict mapping (parent, child) values to the score breakdown for the
+                cardinality distributions of that foreign key.
+        """
+        if set(real_data.keys()) != set(synthetic_data.keys()):
+            raise ValueError('`real_data` and `synthetic_data` must have the same tables.')
+        if metadata is None:
+            raise ValueError('`metadata` cannot be ``None``.')
+        if not isinstance(metadata, dict):
+            metadata = metadata.to_dict()
+
+        score_breakdowns = {}
+        for table_name, table in metadata['tables'].items():
+            for field_name, field in table['fields'].items():
+                if 'ref' in field:
+                    parent_table_name = field['ref']['table']
+                    parent_field_name = field['ref']['field']
+                    cardinality_real = get_cardinality_distribution(
+                        real_data[parent_table_name][parent_field_name],
+                        real_data[table_name][field_name],
+                    )
+                    cardinality_synthetic = get_cardinality_distribution(
+                        synthetic_data[parent_table_name][parent_field_name],
+                        synthetic_data[table_name][field_name],
+                    )
+                    statistic, _ = ks_2samp(cardinality_real, cardinality_synthetic)
+                    score_breakdowns[(parent_table_name, table_name)] = {'score': 1 - statistic}
+
+        if len(score_breakdowns) == 0:
+            return {'score': np.nan}
+
+        return score_breakdowns
+
+    @classmethod
+    def compute(cls, real_data, synthetic_data, metadata=None):
+        """Compute the average of cardinality shape similarity in the given tables.
+
+        Compute the average shape similarity in cardinality distributions for
+        (parent, child) relationship in the given tables, as specified in the metadata.
+        The shape similarity is computed based on the KSComplement.
+
+        Args:
+            real_data (dict[str, pandas.DataFrame]):
+                The tables from the real dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            synthetic_data (dict[str, pandas.DataFrame]):
+                The tables from the synthetic dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            metadata (dict):
+                Multi-table metadata dict. If not passed, it is build based on the
+                real_data fields and dtypes.
```

Comment:
Can this default to `None` ? I think we don't build the metadata anywhere. If it's `None` the `compute_breakdown` raises an error.